### PR TITLE
lp-172 : add dependency google-java-format

### PR DESCRIPTION
**Pull request by lduranteau**
### JIRA link
https://companieshouse.atlassian.net/browse/LP-172

### Change description
Add dependency google-java-format

### Work checklist

* [ ] Tests added where applicable
* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__
### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: .github/PULL_REQUEST-TEMPLATE.md [hunk 1]
```diff
@@ -20,10 +16,14 @@
 * [ ] I have added/updated functional tests where appropriate
 * [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)
 
-__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__
+__An exhaustive list of peer review checks can be
```
Comment (mwestacott) [inline .github/PULL_REQUEST-TEMPLATE.md:19]: Did you intend to push the changes to this template? Nothing to do with the linked ticket...

File: pom.xml [hunk 2]
```diff
@@ -67,6 +67,11 @@
             <artifactId>structured-logging</artifactId>
             <version>${structured-logging.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.google.googlejavaformat</groupId>
+            <artifactId>google-java-format</artifactId>
+            <version>1.24.0</version>
```
Comment (mwestacott) [inline pom.xml:73]: We should define this version number as a property and then reference it here, like for all the other dependencies (e.g. structured-logging).

File: pom.xml [hunk 3]
```diff
@@ -112,7 +117,8 @@
                         <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/ci-corretto-build-21:latest</image>
                     </from>
                     <to>
-                        <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/local/limited-partnerships-api:latest</image>
+                        <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/local/limited-partnerships-api:latest
+                        </image>
```
Comment (mwestacott) [inline pom.xml:121]: I don't think this needed to change - image tag can remain on the same line (120).

File: .github/PULL_REQUEST-TEMPLATE.md (hunk 1 above)
Comment (lduranteau) [reply .github/PULL_REQUEST-TEMPLATE.md:19]: this was done at the pre-commit
all the files recovered during the merge have been reformatted

File: pom.xml (hunk 3 above)
Comment (mwestacott) [reply pom.xml:122]: Is that manual formatting or auto-formatting from Google Java formatter? I would say revert this.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: pom.xml (hunk 3 above)
Comment (mwestacott) [reply pom.xml:122]: But why is a Java formatter changing XML files?

File: config/pre-commit [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+full_path=$(realpath $0)
+dir_path=$(dirname $full_path)
+
+jar_file="$dir_path/google-java-format-1.24.0-all-deps.jar"
```
Comment (markpit) [inline config/pre-commit:6]: could you reference the jar file downloaded by maven instead? then we wouldn't have to commit the jar file, something like
~/.m2/repository/com/google/googlejavaformat/google-java-format/1.24.0/google-java-format-1.24.0.jar

File: config/setup [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+full_path=$(realpath $0)
+dir_path=$(dirname $full_path)
+
+# The google-java-format version must match the one used in the pom.xml file
+version="1.24.0"
```
Comment (markpit) [inline config/setup:7]: you might also be able to read the version from the pom using mvn help:evaluate
VERSION=$(mvn help:evaluate -Dexpression=google-java-format.version -q -DforceStdout)

File: pom.xml (hunk 3 above)
Comment (lduranteau) [reply pom.xml:122]: I don't see why it is a problem

File: config/pre-commit (hunk 4 above)
Comment (lduranteau) [reply config/pre-commit:6]: We can't use the script from .m2 because it needs the -all-deps.jar

**Review by markpit — APPROVED** [review]

File: config/setup (hunk 5 above)
Comment (lduranteau) [reply config/setup]: use version form pom.xml

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: config/setup [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
```
Comment (markpit) [inline config/setup:2]: just had another idea, could this script check if the google-java-format-1.24.0-all-deps.jar exists and of not, download it with curl - curl -LJO https://github.com/google/google-java-format/releases/download/v1.24.0/google-java-format-1.24.0-all-deps.jar

Comment (markpit) [reply config/setup:2]: then we can remove the jar from our git repo

Comment (markpit) [reply config/setup:2]: and use the version var in the curl

Comment (lduranteau) [reply config/setup:2]: updated
download jar file, copy to hooks folder, remove it from config

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request closed without merging**